app/src/main/java/com/sdp/swiftwallet/presentation/friend/FriendFragment.java

```diff
@@ -0,0 +1,145 @@
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.SetOptions;
+import com.sdp.cryptowalletapp.R;
+import com.sdp.swiftwallet.common.FirebaseUtil;
+import com.sdp.swiftwallet.presentation.friend.fragments.FriendItem;
+import com.sdp.swiftwallet.presentation.friend.fragments.FriendItemFragment;
+import com.sdp.swiftwallet.presentation.friend.fragments.FriendItemRecyclerViewAdapter;
+import com.sdp.swiftwallet.presentation.wallet.fragments.WalletItemFragment;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+
+public class FriendFragment extends Fragment{
+    private FirebaseAuth mAuth;
+    private FirebaseFirestore db;
+    private FriendItemFragment friendItemFragment;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        mAuth = FirebaseUtil.getAuth();
+        db = FirebaseFirestore.getInstance();
+
+        friendItemFragment = new FriendItemFragment();
+        getChildFragmentManager().beginTransaction()
+                .add(R.id.friend_nested_frag_container, friendItemFragment, FriendItemFragment.class.getName())
+                .setReorderingAllowed(true)
+                .commit();
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
+        View view = inflater.inflate(R.layout.fragment_friend, container, false);
+
+        Button showAddFriendButton = view.findViewById(R.id.showAddFriendBtn);
+        showAddFriendButton.setOnClickListener(v -> {
+            if (view.findViewById(R.id.friendCode).getVisibility() != View.VISIBLE) {
+                view.findViewById(R.id.friendCode).setVisibility(View.VISIBLE);
+                view.findViewById(R.id.addFriendBtn).setVisibility(View.VISIBLE);
+                view.findViewById(R.id.addFriendStatus).setVisibility(View.GONE);
+            } else {
+                view.findViewById(R.id.friendCode).setVisibility(View.GONE);
+                view.findViewById(R.id.addFriendBtn).setVisibility(View.GONE);
+                view.findViewById(R.id.addFriendStatus).setVisibility(View.GONE);
+            }
+        });
+
+        Button addFriendButton = view.findViewById(R.id.addFriendBtn);
+        addFriendButton.setOnClickListener(v -> db.collection("friend_list")
+                .document(Objects.requireNonNull(mAuth.getCurrentUser()).getUid()).get()
+                .addOnCompleteListener(task -> {
+                    if (task.isSuccessful()) {
+                        DocumentSnapshot document = task.getResult();
```

Comment:
Would it be possible to refactor most of this into other methods? Would make it more readable imo